caper/resource_analysis.py

```diff
@@ -0,0 +1,338 @@
+            key = '{reduce_name}({vars})'.format(
+                reduce_name=reduce_in_file_vars.__name__,
+                vars=','.join(sorted(x_data.keys())),
+            )
+            # transpose to reduce(sum by default) file sizes
+            # over all in_file_vars
+            tranposed = np.transpose([vec for vec in x_data.values()])
+            reduced = [reduce_in_file_vars(vec) for vec in tranposed]
+            x_data = {key: reduced}
+
+        # tranpose it to make x matrix
+        x_matrix = np.transpose([vec for vec in x_data.values()])
+
+        result = {'x': x_data, 'y': y_data, 'coeffs': {}}
+        for res_metric, y_vec in y_data.items():
+            result['coeffs'][res_metric] = self._solve(
+                x_matrix=x_matrix,
+                y_vec=y_vec,
+                plot_y_label=res_metric,
+                plot_title=task_name,
+                plot_pp=plot_pp,
+            )
+
+        # a bit hacky way to recursively convert numpy type into python type
+        json_str = json.dumps(result, default=convert_type_np_to_py)
+        return json.loads(json_str)
+
+    def _get_all_task_names(self):
+        """Get all task names.
+        """
+        return list(set([task['task_name'] for task in self._tasks]))
+
+    @abstractmethod
+    def _solve(self, x_matrix, y_vec, plot_y_label=None, plot_title=None, plot_pp=None):
+        raise NotImplementedError
+
+
+class LinearResourceAnalysis(ResourceAnalysis):
+    def _solve(self, x_matrix, y_vec, plot_y_label=None, plot_title=None, plot_pp=None):
+        """Solve y = A(X) with linear regression.
+        Also make a scatter plot (for one-dimensional x_matrix only).
+        Use `reduce_in_file_vars` in ResourceAnalysis.analyze()
+        to reduce a matrix into a vector.
+
+        Args:
+            x_matrix:
+                X Matrix.
+            y_vec:
+                y vector.
+            plot_y_label:
+                y label for plot.
+            plot_title:
+                Plot title.
+            plot_pp:
+                Matplotlib's PDF backend PdfPages object.
+        Returns:
+            Tuple of (coeffs, intercept).
+        """
+        import numpy as np
+        from sklearn import linear_model
+
+        x_matrix = np.array(x_matrix)
+
+        try:
+            model = linear_model.LinearRegression().fit(x_matrix, y_vec)
+            coeffs, intercept = model.coef_, model.intercept_
+
+        except (TypeError, ValueError):
+            logger.error(
+                'Failed to solve due to type/dim mismatch. '
+                'Too few data or invalid resource monitoring script? '
+                'title: {title}, y_label: {y_label}, '
+                'y_vec={y_vec}, x_matrix: {x_matrix}'.format(
+                    title=plot_title,
+                    y_label=plot_y_label,
+                    y_vec=y_vec,
+                    x_matrix=x_matrix,
+                )
+            )
+            return
```

Comment:
Seems like this could mask other errors, and the message would be unhelpful in those cases. You should at least try to preserve the original traceback with `exc_info=True` or using `logger.exception` instead of `logger.error`.

Reply:
Yes, added `exc_info=True` and removed `TypeError` from catch-list. It will only catch `ValueError`.